runner/operator/access/v1_0_0/merge_fastqs/__init__.py

```diff
@@ -0,0 +1,123 @@
+    'dnaInputNg',
+    'captureInputNg',
+    'baitSet',
+    'sex',
+    'barcodeIndex',
+    'libraryVolume',
+    'captureName',
+    'libraryConcentrationNgul',
+    'captureConcentrationNm'
+]
+def construct_inputs(samples):
+    with open(os.path.join(WORKDIR, 'input_template.json.jinja2')) as file:
+        template = Template(file.read())
+
+
+    samples = list(group_by_sample_id(samples).values())
+
+    inputs = list()
+    for sample_files in samples:
+        fastqs = group_by_fastq(sample_files)
+        metadata = sample_files[0]["metadata"]
+        metadata.update({
+            "dnaInputNg": calc_avg(sample_files, "dnaInputNg"),
+            "captureInputNg": calc_avg(sample_files, "captureInputNg"),
+            "libraryVolume": calc_avg(sample_files, "libraryVolume"),
+            "libraryConcentrationNgul": calc_avg(sample_files, "libraryConcentrationNgul"),
+            "captureConcentrationNm": calc_avg(sample_files, "captureConcentrationNm")
+
+        })
+
+        fastq1s = [{
+            "class": "File",
+            "location": "juno://" + fastq["path"]
+        } for fastq in fastqs["R1"]]
+
+        fastq2s = [{
+            "class": "File",
+            "location": "juno://" + fastq["path"]
+        } for fastq in fastqs["R2"]]
+
+        input_file = template.render(
+            fastq1_files=json.dumps(fastq1s),
+            fastq2_files=json.dumps(fastq2s),
+        )
+
+        inputs.append((json.loads(input_file), metadata))
+
+    return inputs
+
+class AccessLegacyFastqMergeOperator(Operator):
+    def get_jobs(self):
+        files = FileRepository.filter(queryset=self.files,
+                                      metadata={'requestId': self.request_id,
+                                                'igocomplete': True})
+        data = [
+            {
+                "id": f.file.id,
+                "path": f.file.path,
+                "file_name": f.file.file_name,
+                "metadata": f.metadata
+            } for f in files
+        ]
+
+        inputs = construct_inputs(data)
+
+        number_of_inputs = len(inputs)
+
+        return [
+            (
+                APIRunCreateSerializer(
+                    data={
+                        'name': "LEGACY FASTQ Merge: %s, %i of %i" % (self.request_id, i + 1, number_of_inputs),
+                        'app': self.get_pipeline_id(),
+                        'output_metadata': {key: metadata[key] for key in METADATA_OUTPUT_FIELDS if
+                                            key in metadata},
+                        'inputs': job,
+                        'tags': {'requestId': self.request_id, 'sampleId': metadata["sampleId"]}}
+                ),
+                job
+             )
+
+            for i, (job, metadata) in enumerate(inputs)
+        ]
+
+def calc_avg(sample_files, field):
+    return sum([float(s["metadata"][field] or 0) for s in sample_files if field in s["metadata"]])/float(len(sample_files))
```

Comment:
Maybe use the number of available instances in the denominator:
`field_count = len(list(filter(lambda s: s["metadata"][field], sample_files)))`
...
`... / float(len(field_count)`